File: src/jv/abstractFactory/Principal.java
package jv.abstractFactory;

public class Principal {
    public static void main(String[] args) {
        Hamburgaria h1 = new Hamburgaria();
        Hamburgaria h2 = new Hamburgaria();
        Hamburgaria h3 = new Hamburgaria();
        h1.criarHamburguer("minas", "completo");
        h2.criarHamburguer("goias", "extra");
        h3.criarHamburguer("goias", "básico");
        h1.criarHamburguer("minas", "completo");

        h1.criarHamburguer("goias", "extra");

        //Tentei prof.

    }
}

File: src/abstractFactory/Principal.kt
package abstractFactory

/* Created by Dhionatã on 05/02/2022 */
class Principal {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val h1 = Hamburgaria()
            val h2 = Hamburgaria()
            val h3 = Hamburgaria()
            h1.criarHamburguer("minas", "completo")
            h2.criarHamburguer("goias", "extra")
            h3.criarHamburguer("goias", "básico")
        }
    }
}